compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/SyntheticAccessorLowering.kt

```diff
@@ -49,6 +55,53 @@ private class SyntheticAccessorLowering(val context: JvmBackendContext) : IrElem
 
     override fun lower(irFile: IrFile) {
         inlineLambdaToCallSite.putAll(InlineReferenceLocator.scan(context, irFile).lambdaToCallSite)
+
+        // Unconditionally add bridges for hidden constructors
+        irFile.transformChildrenVoid(object: IrElementTransformerVoid() {
+            private val hiddenDeclarations = mutableSetOf<IrConstructor>()
+
+            private fun handleConstructor(declaration: IrConstructor): IrConstructorSymbol? {
```

Comment:
Could you explain the change in this function in the commit "JVM IR: Don't compute signatures in CallableReferenceLowering"? I don't quite get why do we keep the `hiddenDeclarations` set, if `functionMap.getOrPut` already guarantees that we don't create more than one accessor per constructor

Reply:
Right, I refactored the code to reuse `functionMap` for this purpose - the problem is that we change the visibility of the constructor to private, after which they no longer have to be hidden.

However, if we assume that the `functionMap` is empty before we generate bridges for hidden constructors then we can use it to determine if we already generated a bridge.